File: Server/src/main/java/com/chat/server/Security.java
package com.chat.server;

import java.security.*;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

import sun.misc.*;

public class Security {
	private static final String ALGO = "AES";
	private static final byte[] keyValue = "TheBestSecretKey".getBytes();

	public static String encrypt(String Data) throws Exception {
		try{
		Key key = new SecretKeySpec(keyValue, ALGO);
        Cipher c = Cipher.getInstance(ALGO);
        c.init(Cipher.ENCRYPT_MODE, key);
        byte[] encVal = c.doFinal(Data.getBytes());
        String encryptedValue = new BASE64Encoder().encode(encVal);
        return encryptedValue;
		}catch(Exception e){System.out.println("encryption error: "+e); return "";}
    }

    public static String decrypt(String encryptedData) throws Exception {
    	try{
    	Key key = new SecretKeySpec(keyValue, ALGO);
        Cipher c = Cipher.getInstance(ALGO);
        c.init(Cipher.DECRYPT_MODE, key);
        byte[] decodedValue = new BASE64Decoder().decodeBuffer(encryptedData);
        byte[] decValue = c.doFinal(decodedValue);
        String decryptedValue = new String(decValue);
        return decryptedValue;
    	}catch(Exception e){System.out.println("decryption error: "+e); return "";}
    }
}

File: Client/Desktop/bin/Security.rb
include Java

java_import Java::javax.crypto.Cipher
java_import Java::javax.crypto.spec.SecretKeySpec
java_import Java::sun.misc.BASE64Encoder
java_import Java::sun.misc.BASE64Decoder

class Security
  @key = "TheBestSecretKey"

  def self.encrypt(data)
    aes = javax.crypto.spec.SecretKeySpec.new(@key.to_java_bytes, "AES")
    cipher = javax.crypto.Cipher.getInstance("AES")
    cipher.init(javax.crypto.Cipher::ENCRYPT_MODE, aes)
    bytes = cipher.doFinal(data.to_java_bytes)
    encryptedValue = BASE64Encoder.new.encode(bytes)
    encryptedValue.gsub!("\r\n", "")
    return encryptedValue.to_java_string
  end

  def self.decrypt(encryptedData)
    aes = javax.crypto.spec.SecretKeySpec.new(@key.to_java_bytes, "AES")
    cipher = javax.crypto.Cipher.getInstance("AES")
    cipher.init(javax.crypto.Cipher::DECRYPT_MODE, aes)
    decodedValue = BASE64Decoder.new.decodeBuffer(encryptedData)
    decValue = cipher.doFinal(decodedValue)
    return decValue.to_s
  end
end